BrickKit/bricks/src/main/java/com/wayfair/brickkit/viewholder/factory/BrickViewHolderFactory.java

```diff
@@ -0,0 +1,114 @@
+                viewHolder = createViewHolderWithViewType(
+                        data.getParent(),
+                        data.getViewType(),
+                        data.getBrickProvider());
+            }
+        } catch (AssertionError ae) {
+            // This shouldn't happen
+            Log.wtf(data.getLogTag(), "Unable to get the layout inflater. ");
+        } catch (Resources.NotFoundException nfe) {
+            Log.w(data.getLogTag(), "Unable to find the resource. ", nfe);
+        } catch (NullPointerException npe) {
+            Log.w(data.getLogTag(), "A brick or the context is null and shouldn't be.");
+        }
+
+        // Since the view holder could be null, if it is, create an empty version and return
+        // its reference instead of returning a null object reference.
+        return null == viewHolder ? createEmptyBrickViewHolder(context) : viewHolder;
+    }
+
+    /**
+     * Creates an {@link com.wayfair.brickkit.viewholder.EmptyBrickViewHolder }
+     *
+     * @param context used to create the {@link EmptyBrickView}
+     * @return the newly created EmptyBrickViewHolder
+     */
+    /* private package */
+    BrickViewHolder createEmptyBrickViewHolder(@NonNull Context context) {
+        EmptyBrickView emptyBrickView = new EmptyBrickView(context);
+        return new EmptyBrickViewHolder(emptyBrickView);
+    }
+
+    /**
+     * Creates a {@link BrickViewHolder} based on the {@param #viewType}.
+     *
+     * @param parent   of the view, passed into the {@link BrickViewHolder}'s constructor.
+     * @param viewType of the view to create.  It can be a @LayoutRes id.
+     * @param provider provides bricks, based on resource id
+     * @return a newly created {@link BrickViewHolder}
+     * @throws AssertionError              thrown when the layout inflater can't be created
+     * @throws Resources.NotFoundException when the view type is zero or the resource cannot be located
+     * @throws NullPointerException        in cases, such as brick creation. (Just a fail safe)
+     */
+    /* private package */
+    BrickViewHolder createViewHolderWithViewType(@NonNull ViewGroup parent,
+                                                 int viewType,
+                                                 @NonNull BrickProvider provider)
+            throws AssertionError, Resources.NotFoundException, NullPointerException {
+
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+        View itemView = inflater.inflate(viewType, parent, false);
+        BaseBrick brick = provider.brickWithLayout(viewType);
+        brick = brick == null ? provider.brickWithPlaceholderLayout(viewType) : brick;
+
+        // Since the brick could be null, if it is, create an empty brick view holder and
+        // return its reference to avoid null pointer exceptions being thrown.
+        @SuppressWarnings("UnnecessaryLocalVariable") // set for debugging w/ breakpoints
```

Comment:
not sure you need the breakpoint code in...if people need to test later, they can modify the code to test...

Reply:
Fixed.